engine/src/components/component.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * component.cpp
+ *
+ * Copyright (c) 2001-2002 Daniel Horn
+ * Copyright (c) 2002-2019 pyramid3d and other Vega Strike Contributors
+ * Copyright (c) 2019-2023 Stephen G. Tuggy, Benjamen R. Meyer, Roy Falk and other Vega Strike Contributors
+ *
+ * https://github.com/vegastrike/Vega-Strike-Engine-Source
+ *
+ * This file is part of Vega Strike.
+ *
+ * Vega Strike is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Vega Strike is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Vega Strike. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// -*- mode: c++; c-basic-offset: 4; indent-tabs-mode: nil -*-
+
+#include "component.h"
+
+#include <random>
+
+Component::Component()
+{
+
+}
+
+
+double random20()
+{
+    if(std::rand() < 0.2) {
+        return std::rand();
+    }
+
+    return 1.0;
```

Comment:
this should probably be:
```c++
double random_value = std::rand();
if (random_value < 0.2) {
    return random_value;
}
return 1.0;
```
As it is, if L40 passes then a new random number will be used and it could still be greater than or equal to 0.2, thereby likely defeating the check.

Reply:
@BenjamenMeyer You are correct, AFAICT.